clinica/pydra/statistics_volume/pipeline.py

```diff
@@ -0,0 +1,241 @@
+import pydra
+from nipype.algorithms.misc import Gunzip
+from pydra import Workflow
+from pydra.tasks.nipype1.utils import Nipype1Task
+
+from clinica.pydra.engine import clinica_io
+
+
+def _check_pipeline_parameters(parameters: dict) -> dict:
+    """Check the parameters passed to the pipeline.
+
+    Parameters
+    ----------
+    parameters : dict
+        Dictionary of parameters to analyze.
+
+    Returns
+    -------
+    dict :
+        Cleaned dictionary of parameters.
+    """
+    from clinica.utils.exceptions import ClinicaException
+
+    # PET-Volume pipeline
+    if parameters["orig_input_data_volume"] == "pet-volume":
+        for param in ("acq_label", "suvr_reference_region"):
+            if not parameters[param]:
+                raise ClinicaException(
+                    f"You selected pet-volume pipeline without setting --{param} flag. "
+                    "Clinica will now exit."
+                )
+
+    # Custom pipeline
+    if parameters["orig_input_data_volume"] == "custom-pipeline":
+        if not all([parameters["custom_file"], parameters["measure_label"]]):
+            raise ClinicaException(
+                "You must set --measure_label and --custom_file flags."
+            )
+        if not parameters["custom_file"]:
+            raise ClinicaException(
+                "Custom pipeline was selected but no 'custom_file' was specified."
+            )
+    return parameters
+
+
+def _build_query(parameters: dict) -> dict:
+    input_name = parameters["orig_input_data_volume"].replace("-", "_")
+    query = {
+        "group_label": parameters["group_label_dartel"],
+        "fwhm": parameters["full_width_at_half_maximum"],
+    }
+    if input_name == "pet_volume":
+        query.update(
+            {
+                "acq_label": parameters["acq_label"],
+                "suvr_reference_region": parameters["suvr_reference_region"],
+                "use_brainmasked_image": True,
+                "use_pvc_data": parameters["use_pvc_data"],
+            }
+        )
+    elif input_name == "t1_volume":
+        query.update({"tissue_number": 1, "modulation": True})
+
+    elif input_name == "custom_pipeline":
+        query = {
+            "pattern": parameters["custom_file"],
+            "description": "custom file provided by user",
+        }
+    return parameters
```

Comment:
Here's why I was getting the error yesterday. The query was the pull parameter dictionary, so the CAPSReader was unable to process it.

```suggestion
    return query
```